### DISCUSSION TITLE
Create another table with PK and string to be a FK in the users table?

### DISCUSSION THREAD

**Discussion by hsul-git** [fastapi-users/fastapi-users#1127] [Q&A]
Is it possible to create a new table in the database (sqlite) besides User with a PK and String?
The PK of the other table should be the FK of the User table and this has the be delivered to the default FastAPI-Users api endpoints like update User, create User etc..

How can i archive this?

This is what i tried to first create the database:

```py
from fastapi import Depends
from typing import AsyncGenerator

from sqlalchemy import Column, ForeignKey, String

from app.config import get_settings
from sqlalchemy.orm import sessionmaker
from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
from sqlalchemy.ext.declarative import DeclarativeMeta, declarative_base
from fastapi_users.db import SQLAlchemyBaseUserTableUUID, SQLAlchemyUserDatabase

settings = get_settings()

user_db_connection_string = settings.user_db_connection_string
Base: DeclarativeMeta = declarative_base()


class User(SQLAlchemyBaseUserTableUUID, Base):
    neo4juser = Column("neo4juser", ForeignKey('OtherTable.pkid'),  primary_key=True)


class OtherTable(SQLAlchemyBaseUserTableUUID, Base):
    pkid = Column('pkid', String(20))
    text = Column('text', String(64))


engine = create_async_engine(user_db_connection_string)
async_session_maker = sessionmaker(engine, class_=AsyncSession, expire_on_commit=False)


async def create_db_and_tables():
    async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.create_all)


async def get_async_session() -> AsyncGenerator[AsyncSession, None]:
    async with async_session_maker() as session:
        yield session


async def get_user_db(session: AsyncSession = Depends(get_async_session)):
    yield SQLAlchemyUserDatabase(session, User, OtherTable)
```

It tells me "Table 'user' is already defined for this MetaData instance.  Specify 'extend_existing=True' to redefine options and columns on an existing Table object.".

Where do i define it? And is it even correct how i try to solve this?

Thank you for helping.

**Comment by frankie567** [comment]
The problem is here:

```py
class OtherTable(SQLAlchemyBaseUserTableUUID, Base):
    pkid = Column('pkid', String(20))
    text = Column('text', String(64))
```

`OtherTable` shouldn't inherit from `SQLAlchemyBaseUserTableUUID`, it's only for the User table to provide the base columns. Otherwise, SQLAlchemy understands you want to define a table named exactly the same, hence the error.

Besides, I don't think you should define yoru foreign key `neo4juser` as primary key. If you want to ensure uniqueness across the table, you should rather use `unique=True`.

**Reply by hsul-git** [reply]
Should i create a seperate table how its displayed in the sqlamlchemy docu? But how do i deliver the additional data and the data from the new table from the database to the api endpoint via Fastapi-Users? The new column has to be set by to a specific value at start and should be able to altered trough User Update method.

Is this actually possible with Fastapi-Users?
